exporter/main.py

```diff
@@ -303,15 +303,40 @@ def match(course):
 
 def get_all_courses(**kwargs):
     log.info('Retrieving all courses')
-
+    # These are options that don't start with the keyword django
+    other_options = ('lms_config', 'studio_config', 'time_constraint')
     # make a set of fixed arguments, so we can memoize
     kwargs = {
         k: v for k, v in kwargs.iteritems()
-        if k.startswith('django') or k == 'lms_config' or k == 'studio_config'
+        if k.startswith('django') or any(k == opt for opt in other_options)
     }
     kwargs['dry_run'] = False  # always query for course names
     kwargs['limit'] = False  # don't limit number of courses
 
+    # Extract the time constraint option and calculate the end date that
+    # should be used to limit the list of returned course IDs.
+    # Please note that the default time constraint is 3 years
+    constraint = str(kwargs.get('time_constraint', '3'))
+    try:
+        constraint = int(constraint)
+    except ValueError as e:
+        # Not sure that this is great idea, but for now, we report the error
+        warn_msg = (
+            'The given time constraint value {c} is not a valid integer. '
+            'The default value of 3 will be used instead: {e}'
+        )
+        log.warning(warn_msg.format(c=constraint, e=e))
+        constraint = 3
+
+    # Calculate the end date by changing today's year to 3 years ago.
+    # Note that using str on a date object returns a date string with the
+    # format %Y-%m-%d
+    today = datetime.date.today()
+    kwargs['end'] = str(today.replace(year=today.year - constraint))
```

Comment:
are you sure this technique always works?  What about leap year dates?  should we be using datetime math instead?
```suggestion
    kwargs['end'] = str(today - datetime.timedelta(days=(365 * constraint)))
```

Reply:
I didn't want to deal with figuring whether to use `365` or `366`. Replacing the `year` component with `year - constraint` seemed to make more sense. But I am happy to use `str(today - datetime.timedelta(days=(365 * constraint)))`.

Reply:
right, but we're just going for approximately 3 years, and at least we know for sure that it works for every possible date so I still stand by my proposal